astropy/stats/bayesian_blocks.py

```diff
@@ -465,24 +497,39 @@ class RegularEvents(FitnessFunc):
         False alarm probability, used to compute the prior on :math:`N_{\rm
         blocks}` (see eq. 21 of Scargle 2013). If gamma is specified, p0 is
         ignored.
+    gamma : float, optional
+        If specified, then use this gamma to compute the general prior form,
+        :math:`p \sim {\tt gamma}^{N_{\rm blocks}}`.  If gamma is specified, p0
+        is ignored.
     ncp_prior : float, optional
         If specified, use the value of ``ncp_prior`` to compute the prior as
         above, using the definition :math:`{\tt ncp\_prior} = -\ln({\tt
         gamma})`.  If ``ncp_prior`` is specified, ``gamma`` and ``p0`` are
         ignored.
     """
 
-    def __init__(self, dt, p0=0.05, gamma=None, ncp_prior=None):
+    def __init__(
+        self,
+        dt: float,
+        p0: float | None = 0.05,
+        gamma: float | None = None,
+        ncp_prior: float | None = None,
+    ) -> None:
         self.dt = dt
         super().__init__(p0, gamma, ncp_prior)
 
-    def validate_input(self, t, x, sigma):
+    def validate_input(
+        self,
+        t: ArrayLike,
+        x: ArrayLike | None = None,
+        sigma: float | ArrayLike | None = None,
+    ) -> tuple[ArrayLike, float | ArrayLike, float | ArrayLike]:
```

Comment:
And this.

Reply:
Fixed in [51ffa26](https://github.com/astropy/astropy/pull/16562/commits/51ffa268c9eb79ebce85b9b56fca7ff92d4db8e2)